tools/Config.mk

```diff
@@ -196,6 +196,26 @@ OBJPATH ?= .
 %.ddh: %.c
 	$(Q) $(MKDEP) --obj-path $(OBJPATH) --obj-suffix $(OBJEXT) $(DEPPATH) "$(CC)" -- $(HOSTCFLAGS) -- $< > $@
 
+# MAX_ARGS - Split a single command up into multiple commands based on maximum number
+#   of words in the command arguments.  This is especially useful for cases like assembling
+#   a large archive or deleting a large number of objects.  The Argument list too long error
+#   is based on the number of bytes in the argument so this is not an exact fix, but selecting
+#   a reasonable number for the maximum number of words can still allow for multiple files
+#   to be supplied to a command but also split it up in the large cases.
+# Example: @$(call MAX_ARGS,ar rcs mylib.a ,2,$(OBJS))
+
+define MAX_ARGS
+  $(eval _args:=)
+  $(foreach obj,$3,$(eval _args+=$(obj))$(if $(word $2,$(_args)),$1$(_args)$(EOL)$(eval _args:=)))
+  $(if $(_args),$1$(_args))
+endef
+define EOL
+
+
+
+
+endef
```

Comment:
I'm not following the purpose of this

Reply:
This is a macro so you need to have the EOL or this will be resolved as 
`ar rcSP fooo.a obj1.o obj2.o ar rcSP fooo.a obj3.o obj4.o` instead of
```
ar rcSP fooo.a obj1.o obj2.o
ar rcSP fooo.a obj3.o obj4.o
```

Reply:
oh ok. maybe add a comment to explain the apparently empty macro?

Reply:
Added a comment about this